### DISCUSSION TITLE
All deprecations are now displayed

### DISCUSSION THREAD

**Discussion by sebastien-jj** [symfony/symfony#58782] [Q&A]
Hi @nicolas-grekas,
I writing to you because you added `error_reporting(-1);` in the `SymfonyErrorHandler.php` file.
Why did you do that please ? Is there a solution to disable this please ?
Thanks for your help.

**Comment by nicolas-grekas** [comment]
SymfonyErrorHandler also registers the ErrorHandler class, which is then the one responsible for handling deprecations.

**Reply by sebastien-jj** [reply]
So what can i do please ?

**Reply by vmoulinier** [reply]
Hello,

Previously, in vendor/symfony/runtime/GenericRuntime.php, you had:

`if ($debug) -> $errorHandler::register($debug);`

and now:

`if (false !== $errorHandler = ($options['error_handler'] ?? BasicErrorHandler::class)) -> $errorHandler::register($debug);`

Debugging is no longer dependent on the debug parameter. Now, errors are logged regardless of the environment, whether dev, preprod, or prod.

The only way to avoid activating errorHandler's register is to set $options['error_handler'] = false;.

Do you have a clean solution to prevent deprecation notices from being logged in the production environment?

Thanks.

**Reply by nicolas-grekas** [reply]
This reminds me https://github.com/symfony/symfony/issues/35575

**Reply by vmoulinier** [reply]
talking about your last commit : https://github.com/symfony/runtime/commit/51477fd01538fccdf9bf8440f1cf985187cdb264

this is not configurable anymore

**Comment by sebastien-jj** [comment]
I found a solution by adding the following code in `public/index.php` : 

```
$_SERVER['APP_RUNTIME_OPTIONS'] = [
    'error_handler' => false,
];

require_once dirname(__DIR__).'/vendor/autoload_runtime.php';
```

This will disable the report of depreciation errors.

**Reply by uncaught** [reply]
Does this not disable any error handling?

**Comment by uncaught** [comment]
Hey @nicolas-grekas, is this change responsible for deprecations showing up on any first request after the php-container was started?

Any time we (re-)start our php-fpm, the first request is greeted with a bunch of PHP deprecations. E.g.:

> Deprecated: Symfony\Bundle\FrameworkBundle\Routing\Router::__construct(): Implicitly marking parameter $context as nullable is deprecated, the explicit nullable type must be used instead in /var/www/html/vendor/symfony/framework-bundle/Routing/Router.php on line 43

This breaks our website on the first load. We (developers) always have to do a second load. Production mode is fine of course.

Further requests are fine, too, thanks to opcache.

Is this the intended behavior and do we have to live with this? The deprecations so far are always in vendor code, out of our reach.

**Reply by stof** [reply]
this deprecation is already fixed in FrameworkBundle. Make sure you use uptodate dependencies.

**Reply by uncaught** [reply]
Yes, I've seen that afterwards. For some reason composer kept the framework at 6.4. I had to explicitly require it to go to 7.

However, that was just an example, we have other deprecations of other libraries up there. Most of those are sadly not as well maintained as Symfony.

**Reply by cvergne** [reply]
Hi @uncaught,
I've found this discussion after encountering a similar problem this week when updating a project from 6.3.1 to 6.4.x, I had PHP deprecations showing up in the response in prod mode (debug disabled).
I figured out it was the `public/index.php` not updated to the [runtime version](https://symfony.com/doc/current/components/runtime.html#usage), and changing the file to the runtime version fixed the bug.
I don't know if it the same problem for you, but in case of, I mention it ;)
